src/nlp/arrow_dataset.py

```diff
@@ -984,22 +1013,26 @@ def filter(
         self,
         function,
         with_indices=False,
+        input_column: Optional[str] = None,
```

Comment:
You need to add `input_column` to the `cache_kwargs` of `filter`.
Same thing for `fn_kwargs`

Reply:
Good point. In the `filter` case, I had to add to add `input_column` to the `fn_kwargs` rather than pass it to `map` as `input_column`, otherwise `map_function` wouldn't be able access the full batches. Not quite as elegant but it works.